File: backend/src/LinearPrecision.Api/Entities/Document.cs
using LinearPrecision.Shared.Domain;

namespace LinearPrecision.Api.Entities;

public class Document : TenantEntity, ISoftDeletable, IAuditable
{
    public string Title { get; set; } = string.Empty;
    public string? Content { get; set; }
    public string ContentFormat { get; set; } = "tiptap-json";
    public Guid? ProjectId { get; set; }
    public Project? Project { get; set; }
    public Guid? ParentDocumentId { get; set; }
    public Document? ParentDocument { get; set; }
    public Guid CreatorId { get; set; }
    public User Creator { get; set; } = null!;
    public bool IsPublished { get; set; }
    public DateTime? PublishedAt { get; set; }
    public int SortOrder { get; set; }

    // ISoftDeletable
    public bool IsDeleted { get; set; }
    public DateTime? DeletedAt { get; set; }
    public Guid? DeletedBy { get; set; }

    // IAuditable
    public Guid? CreatedBy { get; set; }
    public Guid? UpdatedBy { get; set; }

    // Navigation properties
    public ICollection<Document> ChildDocuments { get; set; } = [];
}

File: lib/api/contracts/document.ts
import type { ApiUserBrief } from "./common";

export interface DocumentResponse {
  id: string;
  workspaceId: string;
  title: string;
  content?: string | null;
  contentFormat: string;
  projectId?: string | null;
  parentDocumentId?: string | null;
  creator: ApiUserBrief;
  isPublished: boolean;
  publishedAt?: string | null;
  sortOrder: number;
  childDocuments?: DocumentResponse[] | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateDocumentRequest {
  title: string;
  content?: string | null;
  contentFormat?: string | null;
  projectId?: string | null;
  parentDocumentId?: string | null;
  isPublished?: boolean | null;
  sortOrder?: number | null;
}

export interface UpdateDocumentRequest {
  title: string;
  content?: string | null;
  contentFormat?: string | null;
  projectId?: string | null;
  parentDocumentId?: string | null;
  isPublished?: boolean | null;
  sortOrder?: number | null;
}
